src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3019,13 +3057,42 @@ export class StudyViewPageStore
     ): void {
         trackStudyViewFilterEvent('genomicCategoricalData', this);
 
+        const chart = this._geneSpecificChartMap.get(uniqueKey);
         const dataFilterValues: DataFilterValue[] = getCategoricalFilterValues(
             values
         );
-        this.updateGenomicDataFiltersByValues(
-            uniqueKey,
-            _.cloneDeep(dataFilterValues)
-        );
+        if (chart!.mutationOptionType) {
+            this.updateMutationDataFilters(uniqueKey, [_.cloneDeep(values)]);
+        } else {
+            this.updateGenomicDataFiltersByValues(
+                uniqueKey,
+                _.cloneDeep(dataFilterValues)
+            );
+        }
+    }
+
+    @action.bound
+    updateMutationDataFilters(uniqueKey: string, values: string[][]): void {
```

Comment:
can we make the name "values" more specific?

Reply:
I renamed it to be valueArrays so that we know it's a two dimensional array.